src/time.cpp

```diff
@@ -37,4 +37,24 @@ namespace libtorrent { namespace aux {
 	time_point time_now() { return clock_type::now(); }
 	time_point32 time_now32() { return time_point_cast<seconds32>(clock_type::now()); }
 
+	time_t to_time_t(const time_point32 &tp)
+	{
+		auto hc_now = clock_type::now();
+		auto sys_now = std::chrono::system_clock::now();
+
+		auto r = sys_now + std::chrono::duration_cast<std::chrono::system_clock::duration>(hc_now - tp);
+		return std::chrono::system_clock::to_time_t(r);
+
+	}
+
+	time_point32 from_time_t(std::time_t t)
+	{
+		auto tp = std::chrono::system_clock::from_time_t(t);
+		auto hc_now = std::chrono::high_resolution_clock::now();
+		auto sys_now = std::chrono::system_clock::now();
+
+		auto r = hc_now + std::chrono::duration_cast<std::chrono::system_clock::duration>(sys_now - tp);
+		return  std::chrono::time_point_cast<time_point32::duration>(r);
```

Comment:
this cast seems to be failing on CI

Reply:
I can see one failling for missing std::time_t, can't see anything related to this.

Reply:
https://github.com/arvidn/libtorrent/actions/runs/4155202534/jobs/7188113893#step:6:278

Reply:
I wonder if this could be a limitation in the simulator, or maybe a missing include

Reply:
is "simulations" compile differently? I can't figure out why it's failing.

Reply:
hmm it seems something from libsimulation end, conversion between std::system_clock to simulated_clock

Reply:
I believe it's coming from here: https://github.com/arvidn/libsimulator/blob/master/include/simulator/simulator.hpp#L155